src/components/PopupWithForm.js

```diff
@@ -32,13 +37,11 @@ export default class PopupWithForm extends Popup {
   // event listener to the form and call the setEventListeners() method of the
   // parent class.
   setEventListeners() {
-    this._popupForm.addEventListener("submit", (evt) => {
-      evt.preventDefault();
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", () => {
       const formValues = this._getInputValues();
       this._handleFormSubmit(formValues);
       this._popupForm.reset();
```

Comment:
You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.